bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/ScienceTargetStrategy.scala

```diff
@@ -15,25 +15,29 @@ case class ScienceTargetStrategy(key: AgsStrategyKey, guideProbe: ValidatableGui
 
   // Since the science target is the used as the guide star, success is always guaranteed.
   override def estimate(ctx: ObsContext, mt: MagnitudeTable)(ec: ExecutionContext): Future[AgsStrategy.Estimate] =
-    Future.successful(AgsStrategy.Estimate.GuaranteedSuccess)
+  Future.successful(AgsStrategy.Estimate.GuaranteedSuccess)
 
   override def analyze(ctx: ObsContext, mt: MagnitudeTable, guideProbe: ValidatableGuideProbe, guideStar: SiderealTarget): Option[AgsAnalysis] =
     AgsAnalysis.analysis(ctx, mt, guideProbe, guideStar, probeBands)
 
   override def analyze(ctx: ObsContext, mt: MagnitudeTable): List[AgsAnalysis] =
     AgsAnalysis.analysis(ctx, mt, guideProbe, probeBands).toList
 
-  override def candidates(ctx: ObsContext, mt: MagnitudeTable)(ec: ExecutionContext): Future[List[(GuideProbe, List[SiderealTarget])]] = {
-    val so = ctx.getTargets.getBase.toSiderealTarget(ctx.getSchedulingBlockStart)
-    Future.successful(List((guideProbe, List(so))))
+  private def toSiderealTargets(ctx: ObsContext): List[SiderealTarget] = {
+    val when = ctx.getSchedulingBlockStart
+    val ts   = ctx.getTargets.getAsterism.allSpTargets.map(_.toSiderealTarget(when))
+    ts.list.toList
   }
 
+  override def candidates(ctx: ObsContext, mt: MagnitudeTable)(ec: ExecutionContext): Future[List[(GuideProbe, List[SiderealTarget])]] =
+    Future.successful(List((guideProbe, toSiderealTargets(ctx))))
+
   override def select(ctx: ObsContext, mt: MagnitudeTable)(ec: ExecutionContext): Future[Option[AgsStrategy.Selection]] = {
     // The science target is the guide star, but must be converted from SPTarget to SkyObject.
-    val siderealTarget = ctx.getTargets.getBase.toSiderealTarget(ctx.getSchedulingBlockStart)
-    val posAngle       = ctx.getPositionAngle
-    val assignment     = AgsStrategy.Assignment(guideProbe, siderealTarget)
-    val selection      = AgsStrategy.Selection(posAngle, List(assignment))
+    val siderealTargets = toSiderealTargets(ctx)
+    val assignments     = siderealTargets.map(AgsStrategy.Assignment(guideProbe, _))
+    val posAngle        = ctx.getPositionAngle
+    val selection       = AgsStrategy.Selection(posAngle, assignments)
     Future.successful(Some(selection))
   }
```

Comment:
This algorithm used to assign the probe to the [one and only] base position, coerced into being a sidereal target. This coercion operation is ridiculously sketchy (it turns non-sidereal targets into fixed targets and falls back to a zenith target when coordinates are unknown) but I'm not touching it. The change simply returns an assignment for all science targets rather than assuming one. The selection takes a `List` so I assume this is ok.